mypy/test/data/check-expressions.test

```diff
@@ -1322,8 +1322,10 @@ class str: pass
 def f(x: int) -> None:
     x = yield f('')
     x = 1
+[builtins fixtures/for.py]
 [out]
 main: note: In function "f":
+main:1: error: The return type of a generator function should be "Generator" or one of its supertypes
```

Comment:
As it stands, it's checking both at once.  A common idiom in the tests seems to be purposefully introducing type errors to check the error message.  Testing the Generator function return type enforcement is important, but I don't have much of an opinion on whether it should be done in this function or a separate one.